datahub-graphql-api/src/main/java/com/linkedin/datahub/graphql/resolvers/AuthenticatedResolver.java

```diff
@@ -0,0 +1,31 @@
+package com.linkedin.datahub.graphql.resolvers;
+
+import com.linkedin.datahub.graphql.QueryContext;
+import com.linkedin.datahub.graphql.exception.AuthenticationError;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.DataFetchingEnvironment;
+
+
+/**
+ * Abstract base class that should be extended by any resolver that serves data that should be behind an auth
+ * wall. Simply checks whether the user is currently authenticated.
+ *
+ * In the future, we may choose to introduce additional "role-oriented" resolvers.
+ */
+public abstract class AuthenticatedResolver<T> implements DataFetcher<T> {
+
+    /**
+     * Abstract method to be overridden by subclasses to implement a secured resolver.
+     */
+    protected abstract T authenticatedGet(DataFetchingEnvironment environment) throws Exception;
```

Comment:
And just dropping `Authorized` makes quite a bit of sense since it is just a dummy implementation anyway.

Reply:
I like this suggestion! Going to replace inheritance with composition here. 

And since we aren't yet using authorization logic we can drop and revisit once a pattern emerges